src/v1/internal/connector.js

```diff
@@ -348,7 +365,9 @@ class Connection {
       return;
     }
 
-    log('C', 'ACK_FAILURE');
+    if (this._log.isEnabled()) {
```

Comment:
Are these `.isEnabled()` calls really needed?
They create some noise and makes it harder to read when most of these functions only have a single row of logic.

Wouldn't a noop logger just do nothing with the level calls anyway?

Reply:
These checks prevent code from generating a message string when logging is disabled. For example:

```javascript
this._log.debug(`${this} S: RECORD ${JSON.stringify(msg)}`);
```

will stringify a record and create a message string with `this` and record content even if `this._log` is a no-op logger. The `isEnabled()` check makes code not even allocate all those strings.

This is an optimization that might be relevant in the connection code because it is very hot. I did not use it in other places that are not performance-critical.

Does this make sense?

Reply:
This makes sense, great!